src/Controller/CRUDController.php

```diff
@@ -693,14 +683,10 @@ public function historyAction(Request $request)
     /**
      * View history revision of object.
      *
-     * @param string|null $revision
-     *
      * @throws AccessDeniedException If access is not granted
      * @throws NotFoundHttpException If the object or revision does not exist or the audit reader is not available
-     *
-     * @return Response
      */
-    public function historyViewRevisionAction(Request $request, $revision = null)
+    public function historyViewRevisionAction(Request $request, ?string $revision = null): Response
```

Comment:
I'm not sure what should be type for `$revision`, docbloc suggests `string|null`, but tests are expecting `int`

Reply:
I think that url param are always string.

Two interesting things:
- Why `null` is allowed ? If there is no revision, I don't think this method will work. So I think we should deprecate the `null` value.
- AuditReader are only used in CRUDController. Since $revision will always be a string, I think we can restrict param typehint in `AuditReaderInterface`. We could restrict to `string` or to `string|int` in order to be consistent with the change made here: https://github.com/sonata-project/SonataAdminBundle/blob/master/src/Model/ModelManagerInterface.php#L72

You use `revision` @phansys ; can you help us about this ?

cc @jordisala1991

Reply:
Even if the mapping on the model is declared as `int`, the value this controller is receiving is a string (I think this is always a string when there is no type declaration for the argument). This behavior is related on how the router handles the request, I think.
If you add a type declaration `int $revision`, it's respected and the argument is received as an integer.

Reply:
So we could restrict `AuditReaderInterface` typehint to `string|int`, 
and remove the `null` default value here.

Do you agree ?